cpp/src/ffi_go/ir/encoder.cpp

```diff
@@ -0,0 +1,134 @@
+#include "encoder.h"
+
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <string_view>
+#include <type_traits>
+#include <vector>
+
+#include <clp/components/core/src/ffi/encoding_methods.hpp>
+#include <clp/components/core/src/ffi/ir_stream/decoding_methods.hpp>
+
+#include <ffi_go/defs.h>
+#include <ffi_go/ir/types.hpp>
+#include <ffi_go/types.hpp>
+
+namespace ffi_go::ir {
+using namespace ffi::ir_stream;
+
+namespace {
+    /**
+     * Generic helper for ir_encoder_encode_*_log_message
+     */
+    template <class encoded_var_view_t>
+    auto encode_log_message(
+            StringView log_message,
+            void* ir_encoder,
+            StringView* logtype,
+            encoded_var_view_t* vars,
+            StringView* dict_vars,
+            Int32tSpan* dict_var_end_offsets
+    ) -> int {
+        using encoded_var_t std::conditional<
```

Comment:
```suggestion
        using encoded_var_t = std::conditional<
```
Sorry for missing `=` in the previous comment